File: teaching-assistant/csp-special-judge/csp21-5/spj.js
const fs = require("fs")
const argv = process.argv

try {
  let stdout = fs.readFileSync(argv[3]).toString().trim()
  let userout = fs.readFileSync(argv[4]).toString().trim()
  let stdoutArray = stdout.split("\n")
  let useroutArray = userout.split("\n")

  const len = stdoutArray.length
  if (len !== useroutArray.length) {
    process.exit(6)
  }

  let maxLoss = 0
  for (let i = 0; i < len; ++i) {
    if (maxLoss < 1e-9) {
      process.exit(4)
    }

    if (isNaN(useroutArray[i])) {
      process.exit(6)
    }

    const n = useroutArray[i].trim().split(" ").length
    const m = stdoutArray[i].trim().split(" ").length

    if (n !== m) {
      process.exit(6)
    }

    for (let j = 0; j < n; ++j) {
      const u = parseFloat(useroutArray[i].trim().split(" ")[j])
      const a = parseFloat(stdoutArray[i].trim().split(" ")[j])

      if (Math.abs(a) <= 1) {
        if (Math.abs(u - a) > maxLoss) {
          maxLoss = abs(u - a)
        }
      } else {
        if (abs(a) > 1 && abs(u - a) / abs(a) > maxLoss) {
          maxLoss = abs(u - a) / abs(a)
        }
      }
    }
  }

  process.exit(4)
} catch (e) {
  process.exit(15)
}

File: teaching-assistant/csp-special-judge/csp21-5/spj.py
#!/usr/bin/python3
from ast import main
import sys

out_list = {}

file_in = open(sys.argv[1], "r")
file_u = open(sys.argv[2], "r")
file_a = open(sys.argv[3], "r")


def main():
    maxloss = 0
    while True:
        line_u = file_u.readline()
        line_a = file_a.readline()
        if not line_a and not line_u:
            if maxloss < 1e-9:
                exit(4)
            else:
                exit(6)

        if not line_a:
            exit(6)

        if not line_u:
            exit(6)

        n = len(line_u.strip().split(" "))
        m = len(line_a.strip().split(" "))

        if n != m:
            exit(6)

        for i in range(n):
            u = float(line_u.strip().split(" ")[i])
            a = float(line_a.strip().split(" ")[i])
            
            if abs(a) <= 1:
                if abs(u - a) > maxloss:
                    maxloss = abs(u - a)
            else:
                if abs(a) > 1 and abs(u - a) / abs(a) > maxloss:
                    maxloss = abs(u - a) / abs(a)


if __name__ == "__main__":
    main()
